Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -3544,31 +3544,60 @@ class LowerToAir {
             Value* left = m_value->child(0);
             Value* right = m_value->child(1);
 
-            // UBFIZ Pattern: d = (n & mask) << lsb 
+            // UBFIZ Pattern: d = (n & mask) << lsb
             // Where: mask = (1 << width) - 1
             auto tryAppendUBFIZ = [&] () -> bool {
                 Air::Opcode opcode = opcodeForType(InsertUnsignedBitfieldInZero32, InsertUnsignedBitfieldInZero64, m_value->type());
                 if (!isValidForm(opcode, Arg::Tmp, Arg::Imm, Arg::Imm, Arg::Tmp))
                     return false;
-                if (left->opcode() != BitAnd)
+                if (!imm(right) || right->asInt() < 0)
                     return false;
-
-                Value* nValue = left->child(0);
-                Value* maskValue = left->child(1);
-                if (m_locked.contains(nValue) || !maskValue->hasInt() || !imm(right) || right->asInt() < 0) 
+                if (!canBeInternal(left))
                     return false;
-
                 uint64_t lsb = right->asInt();
-                uint64_t mask = maskValue->asInt();
-                if (!mask || mask & (mask + 1))
+
+                auto doAppend = [&](Value* nValue, uint64_t mask) -> bool {
+                    if (m_locked.contains(nValue))
+                        return false;
+
+                    if (!mask || mask & (mask + 1))
```

Comment:
Nit: Can we add an isPowerOfTwoMinusOne to function to StdLibExtras.h and use that? Remember that's what this is doing took me a bit. Or we could just copy the comment "Where: mask = (1 << width) - 1" from above here.

Reply:
Actually, I prefer having this instead of the function since `isPowerOfTwoMinusOne` is a bit too ambiguous about what it will return. For example, I have no solid answer to the question "should we return true for `UINT64_MAX`?".